soroban-sdk/src/token.rs

```diff
@@ -156,6 +158,46 @@ pub trait Interface {
     /// i128]`
     fn burn_from(env: Env, spender: Address, from: Address, amount: i128);
 
+    /// Returns the number of decimals used to represent amounts of this token.
+    ///
+    /// # Panics
+    ///
+    /// If the contract has not yet been initialized.
+    fn decimals(env: Env) -> u32;
+
+    /// Returns the name for this token.
+    ///
+    /// # Panics
+    ///
+    /// If the contract has not yet been initialized.
+    fn name(env: Env) -> String;
+
+    /// Returns the symbol for this token.
+    ///
+    /// # Panics
+    ///
+    /// If the contract has not yet been initialized.
+    fn symbol(env: Env) -> String;
+}
+
+/// Interface for admin capabilities for Token contracts, such as the Stellar
+/// Asset Contract.
+#[contractspecfn(name = "Spec", export = false)]
+#[contractclient(crate_path = "crate", name = "AdminClient")]
+pub trait AdminInterface {
```

Comment:
Shall we consider removing this completely? I'm not sure if there is a lot of value in standardizing the admin interface TBH, at least not necessarily at SDK level (also stuff like clawback may be better off as optional). I'm ok with this PR for now, just a consideration.

Reply:
I'm also wondering what to do about this. The admin interface is really only here because we need a client in the SDK for calling the stellar asset contract's admin functions.

Instead of having the admin interface here, maybe we move it to a `stellar_asset` module? We've also been trying to keep stellar specific concepts out of the SDK though.

Reply:
I agree this is somethign we should continue to think about and increment on.

Reply:
Yeah, I think there were several discussions where we agreed that it's ok to provide admin functionality just for SAC. So the move would probably make things more clear.

Reply:
Opened issue to continue discussion: https://github.com/stellar/rs-soroban-sdk/issues/1014